components/dashboard/src/App.tsx

```diff
@@ -39,6 +43,10 @@ const App: FC = () => {
     // Setup analytics/tracking
     useAnalyticsTracking();
 
+    if (location.pathname === "/linkedin" && search.get("code") && search.get("state")) {
```

Comment:
Moving this to the top of the render tree so it's not prevented by any blocking flows.